src/main/java/com/limechain/babe/state/EpochDescriptor.java

```diff
@@ -0,0 +1,18 @@
+package com.limechain.babe.state;
+
+import com.limechain.chain.lightsyncstate.BabeEpoch;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.javatuples.Pair;
+
+import java.math.BigInteger;
+
+/**
+ * Represents the BABE constant and secondary slot.
+ */
+@Getter
+@AllArgsConstructor
+public class EpochDescriptor {
+    private Pair<BigInteger, BigInteger> constant;
+    private BabeEpoch.BabeAllowedSlots allowedSlots;
```

Comment:
```suggestion
@Value
public class EpochDescriptor {
    Pair<BigInteger, BigInteger> constant;
    BabeEpoch.BabeAllowedSlots allowedSlots;
```

Reply:
I think this values are going to be changeble in runtime so @Value would provide immutability in this case

Reply:
In the current setup we also have some kind of an immutability, private fields with no setters that are changeable only by creating a new instance.

Reply:
Yeah I planned to add setters while would be working on another task(for state update). Do you want me to add setters now?

Reply:
Ah then it's ok.